File: apps/pos/native-windows/DesktopKitchenPOS/Networking/AuthTokenStore.cs
namespace DesktopKitchenPOS.Networking;

public static class AuthTokenStore
{
    private static string? _token;
    private static readonly object _lock = new();

    public static string? Token
    {
        get { lock (_lock) { return _token; } }
        set { lock (_lock) { _token = value; } }
    }
}

File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/networking/AuthTokenStore.kt
package com.desktopkitchen.pos.networking

import java.util.concurrent.atomic.AtomicReference

object AuthTokenStore {
    private val _token = AtomicReference<String?>(null)

    var token: String?
        get() = _token.get()
        set(value) = _token.set(value)
}
